src/int/i15_montmul.c

```diff
@@ -25,7 +25,7 @@
 #include "inner.h"
 
 /* see inner.h */
-void
+__attribute__((optimize("-O3"))) void
```

Comment:
This `-O3` caused a massive code bloat when I used it in early porting and I found that `-O2` gave almost identical performance without the PROGMEM explosion.  Can you see if this is really necessary in your app?  And if so, can you bracket just the functions that need it?

Reply:
You are right, actually -O2 does not change anything in this function. Dropped.